docs/content/querying/groupbyquery.md

```diff
@@ -106,3 +112,89 @@ your filter, you can use a [filtered dimensionSpec](dimensionspecs.html#filtered
+generally be faster than groupBy. The major differences are that it is implemented in a fully streaming manner (taking
+advantage of the fact that segments are already sorted on time) and does not need to use a hash table for merging.
+
+- For queries with a single "dimensions" element (i.e. grouping by one string dimension), the [TopN query](topnquery.html)
+will sometimes be faster than groupBy. This is especially true if you are ordering by a metric and find approximate
+results acceptable.
+
+#### Nested groupBys
+
+Nested groupBys (dataSource of type "query") are performed the same way for both "v1" and "v2". The broker runs the
+inner groupBy query in the usual way, then materializes the inner query's results, then runs the outer query on those
+materialized results. In particular, the outer query is not distributed at all; it takes place completely on the broker.
+Currently the materialized results are stored on-heap in the broker, and the outer query is done in a single-threaded
+fashion.
+
+#### Server configuration
+
+When using the "v1" strategy, the following runtime properties apply:
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.query.groupBy.defaultStrategy`|Default groupBy query strategy.|v1|
+|`druid.query.groupBy.maxIntermediateRows`|Maximum number of intermediate rows for the per-segment grouping engine. This is a tuning parameter that does not impose a hard limit; rather, it potentially shifts merging work from the per-segment engine to the overall merging index. Queries that exceed this limit will not fail.|50000|
+|`druid.query.groupBy.maxResults`|Maximum number of results. Queries that exceed this limit will fail.|500000|
+|`druid.query.groupBy.singleThreaded`|Merge results using a single thread.|false|
+
+When using the "v2" strategy, the following runtime properties apply:
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.query.groupBy.defaultStrategy`|Default groupBy query strategy.|v1|
+|`druid.query.groupBy.bufferGrouperInitialBuckets`|Initial number of buckets in the off-heap hash table used for grouping results. Set to -1 to use a reasonable default.|-1|
+|`druid.query.groupBy.maxMergingDictionarySize`|Maximum amount of heap space (approximately) to use for the string dictionary during merging. When the dictionary exceeds this size, a spill to disk will be triggered.|25000000|
+|`druid.query.groupBy.maxOnDiskStorage`|Maximum amount of disk space to use, per-query, for spilling result sets to disk when either the merging buffer or the dictionary fills up. Queries that exceed this limit will fail. Set to zero to disable disk spilling.|0 (disabled)|
+
+Additionally, the "v2" strategy uses merging buffers for merging. It is currently the only query implementation that
+does so. By default, Druid is configured without any merging buffer pool, so to use the "v2" strategy you must also
+set `druid.processing.numMergeBuffers` to some non-zero number.
```

Comment:
Also mention that you might need to adjust jvm maxDirectMemory when increasing numMergeBuffers, 
We have seen many people miss that in past with processing threads.